acceptance/tests/external_facts/fact_directory_precedence.rb

```diff
@@ -0,0 +1,89 @@
+test_name "Fact directory precedence and resolution order for facts"
+
+confine :except, :platform => 'windows'
+
+# TestRail: C59201
+
+require 'facter/acceptance/user_fact_utils'
+extend Facter::Acceptance::UserFactUtils
+
+# Generate an external fact dynamically
+def ext_fact(value='BASIC')
+  "test: '#{value}'"
+end
+
+agents.each do |agent|
+  # The directories that facter processes facts
+  os_version = on(agent, facter('kernelmajversion')).stdout.chomp.to_f
+  factsd_dir = get_factsd_dir(agent['platform'], os_version)
+  etc_factsd_dir = get_etc_factsd_dir(agent['platform'], os_version)
+  etc_puppetlabs_factsd_dir = get_etc_puppetlabs_factsd_dir(agent['platform'], os_version)
+  factsd_path = "#{factsd_dir}/test.yaml"
+  etc_factsd_path = "#{etc_factsd_dir}/test.yaml"
+  etc_puppetlabs_factsd_path = "#{etc_puppetlabs_factsd_dir}/test.yaml"
+
+  # ensure the fact directory we want to use exists
+  step "Agent #{agent}: create facts directory (#{etc_puppetlabs_factsd_dir})"
+  on(agent, "rm -rf #{etc_puppetlabs_factsd_dir}")
+  on(agent, "mkdir -p #{etc_puppetlabs_factsd_dir}")
+
+  # A fact in the etc_puppetlabs_factsd_dir directory should resolve to the fact
+  step "Agent #{agent}: create and resolve a custom fact in #{etc_puppetlabs_factsd_dir}"
+  create_remote_file(agent, etc_puppetlabs_factsd_path, ext_fact('etc_puppetlabs_path'))
+  on(agent, facter("test"))
```

Comment:
So in general, I have seen the following idiom to be more common:
```
on(agent, facter("test")) do
    assert_match(/etc_puppetlabs_path/, stdout, "Fact from #{etc_puppetlabs_factsd_dir} did not resolve correctly")
end
```

I think that is preferred so you can print a descriptive message of the failure. The functionality should be largely the same though, so might want to get a second opinion. The same is true for all the following test cases.